ts/addon.ts

```diff
@@ -124,6 +127,27 @@ export default addon({
     }
   },
 
+  _checkAddonAppFiles() {
+    // Emit a warning for addons that are under active development...
+    let isDevelopingAddon = !this.app && (this.parent as Addon).isDevelopingAddon();
+
+    // ...and are at the root of the project (i.e. not in-repo)...
+    let isRootAddon = this.parent.root === this.project.root;
+
+    // ...and have .ts files in their `app` directory.
+    let appDir = `${this.parent.root}/app`;
+    if (isDevelopingAddon && isRootAddon && fs.existsSync(appDir)) {
+      let tsFilesInApp = walkSync(appDir, { globs: ['**/*.ts'] });
+      if (tsFilesInApp.length) {
+        this.ui.writeWarnLine(
+          `found .ts files in ${appDir}\n` +
```

Comment:
Can we elaborate this and/or tweak the wording a little? I'd like for us to make clear that the `.ts` file *won't be compiled*, so e.g. imports will fail.

Reply:
You mean like on the next line where it says we _only_ compile files in `addon/`? 😜

But in seriousness, do you have a suggestion for phrasing here that will make it clearer?

Reply:
Yeah, I kept going back and forth on the wording and whether the next line was clear enough and I kept landing on “somebody is going to fail to understand this.” Still thinking on it; haven’t gotten anywhere. Let’s ship it as is and if one of us comes up with a better phrasing we can update it.